server/webapp/WEB-INF/rails.new/app/presenters/api_v2/base_representer.rb

```diff
@@ -97,11 +97,21 @@ def errors
     end
 
     def to_hash(*options)
-      super.deep_symbolize_keys unless @represented.nil?
+      begin
+        super.deep_symbolize_keys
+      rescue Exception => e
+        Rails.logger.error(e)
+        raise ApiV2::BadRequest, "API Version 2 does not support the contract for this request. Are you sure you're using the right API version?"
```

Comment:
This can be cause if wrong data-types are submitted by the user and passed onto java. For e.g. calling `setName(String)` and the user sends a boolean for the `name` attribute. In which case, this error may not make sense.

Reply:
To not break old APIs, we introduce new properties, different value types, etc. If a user tries to do a get of say, pipeline config API using an old version (v3) for a pipeline that's been configured using say, v6, how do we let the users know that its wrong? It throws 500 currently, we are making it a bad request in this PR. Are you saying the error message should just be `check the logs`?

Reply:
Not quite, I'm suggesting that what you're telling the user may actually not be correct in situations where users submit bad data (mostly unintentionally) and get an error which says something about "not supporting contract".

Reply:
To give a more concrete example:

```
curl -s https://build.gocd.org/go/api/admin/pipelines/ketan-test \
  -H 'Accept: application/vnd.go.cd.v5+json' \
  -XPATCH -d '{"label_template": true, "name": "ketan-test"}' \
  -H'Content-Type: application/json' \
  -H'If-Match: "..."'

<!DOCTYPE html>
<html>
<head>
  <title>We're sorry, but something went wrong (500)</title>
...
</html>
```

Reply:
What do you suggest then as a better error message?

Reply:
What if we said something along the lines of :
```
There is something wrong with your request. This would usually happen if
* there is a API version mismatch
* you submitted an incorrect data-type for one or more elements
Please check logs. 
```
May be not exactly the same words, but something along those lines so the user gets a hint of what things to look at.